src/main/java/st/ratpack/auth/CachingTokenValidator.java

```diff
@@ -22,38 +22,21 @@ public CachingTokenValidator(TokenValidator upstreamValidator) {
 			.maximumSize(10000L)
 			.expireAfterWrite(5L, TimeUnit.MINUTES)
 			//			.executor(Execution.current().getEventLoop())  Don't do this it makes Ratpack hang.
-			.build();
+			.build(this::loadCache);
 	}
 
 	@Override
 	public Promise<Optional<OAuthToken>> validate(String token) {
-		Promise<Optional<OAuthToken>> promiseOAuthToken = cache.getIfPresent(token);
-
-		if (promiseOAuthToken == null) {
-			//Cache miss use the upstream validator
-			logger.trace("Cache MISS: " + token);
-			promiseOAuthToken = upstreamValidator.validate(token);
-
-			//Make sure we only call the validate on the upstream once
-			promiseOAuthToken = promiseOAuthToken.cache();
-
-			promiseOAuthToken.then(optionalOAuth -> {
-				logger.trace("PUTTING: " + optionalOAuth);
-				cache.put(token, Promise.value(optionalOAuth));
```

Comment:
By only adding to the cache on a successful call "then" we make sure its not caching the error cases of the promise.